ea-2021-04-ndvi-automation-nkorinek.py

```diff
@@ -0,0 +1,646 @@
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_ts_single_site.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    single_scene_points += 2
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_ts_single_site.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# YOUR CODE HERE
+
+# Cloud no data vals for Landsat 8
+vals = [328, 392, 840, 904, 1350, 352, 368, 416,
+        432, 480, 864, 880, 928, 944, 992, 480, 992]
+
+
+# Get a list of each directory
+path = os.path.join("ndvi-automation", 
+                    "sites")
+
+# Get a list of each directory
+sites = glob(path + "/*/")
+
+# Define columns and list used to create pandas dataframe
+col_names = ["site", "date", "mean_ndvi"]
+output = []
+
+for site in sites:
+
+    # Get site name
+    site_name = os.path.basename(os.path.normpath(site))
+
+    # Define directories
+    landsat_dir = os.path.join(site, "landsat-crop")
+    vector_dir = os.path.join(site, "vector")
+
+    # Open crop boundary
+    site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+    crop_bound = gpd.read_file(site_boundary_path)
+
+    # Get list of landsat scene directories
+    landsat_dirs = sorted(glob(os.path.join(landsat_dir, "LC08*")))
+    #landsat_dirs.sort()
```

Comment:
Oops  -  looks like you left some commented code in your file @nkorinek

Reply:
Oops, good catch!